### PR TITLE
PS-8442: Fix crash in buf_pool_invalidate_instance()

### PR DESCRIPTION
**Pull request by oleksandr-kachan**
https://jira.percona.com/browse/PS-8442

Revert changes introducing recv_writer_thread logic back in 8.0.30.
This thread is not needed in PS, its job is done by LRU threads.

### CODE REVIEW

File: storage/innobase/log/log0recv.cc [hunk 1]
```diff
@@ -1331,17 +1331,26 @@ dberr_t recv_apply_hashed_log_recs(log_t &log, bool allow_ibuf) {
 
     mutex_exit(&recv_sys->mutex);
 
+    /* Stop the recv_writer thread from issuing any LRU
+    flush batches. */
+    mutex_enter(&recv_sys->writer_mutex);
+
```
Comment (satya-bodapati) [inline storage/innobase/log/log0recv.cc]: Please check about recv_writer_thread() in previous 8.0 versions. 

I think we made a Merge mistake in 8.0.30. We never rely on recv_writer_thread() to do LRU list flushing.
From 8.0.30, we brought upstream logic as is.

During recovery too, our LRU threads will do the flushing.

Please revert the entire recv_writer_thread logic and this change too and test.

If it still fails, the fix should be something else.

**Review by satya-bodapati — CHANGES_REQUESTED** [review]

File: storage/innobase/log/log0recv.cc (hunk 1 above)
Comment (oleksandr-kachan) [reply storage/innobase/log/log0recv.cc]: Removed  recv_writer_thread() related code

**Comment by satya-bodapati** [comment]
@oleksandr-kachan Can you please paste any result that confirms removal of this rec writer thread solves the buf_pool_validate_instance crash?

**Comment by oleksandr-kachan** [comment]
> @oleksandr-kachan Can you please paste any result that confirms removal of this rec writer thread solves the buf_pool_validate_instance crash?

@satya-bodapati, just run previously failing test on Jenkins https://ps80.cd.percona.com/job/percona-server-8.0-pipeline/40888/

**Review by percona-ysorokin — APPROVED** [review]
LGTM

**Pull request merged**